backend/src/main/kotlin/cz/loono/backend/api/service/FirebaseAuthService.kt

```diff
@@ -28,7 +30,28 @@ class FirebaseAuthService : JwtAuthService {
             return JwtAuthService.VerificationResult.Error("Could not verify JWT.")
         }
 
-        return JwtAuthService.VerificationResult.Success(decodedToken.uid)
+        if (decodedToken.email == null) {
+            val error = "Firebase accounts without email are not permitted.\n" +
+                "UID: ${decodedToken.uid}\n" +
+                "Probable reason: Client application may have allowed a login method " +
+                "which doesn't provide user email. Loono only permits login methods which provide user email."
+            logger.error(error)
+
+            return JwtAuthService.VerificationResult.MissingPrimaryEmail
+        }
+
+        if (decodedToken.name == null) {
```

Comment:
Is a name required? We could generate some if not presented like "user1234". Maybe question to Martin. Just asking...

Reply:
oh, that's information from Firebase, so will be filled?

Reply:
We can generate a pseudoname - I just assumed it is required since the "nickname" is an integral part of the user account. We should discuss this with Martin.

Some login providers don't include a name, because there may be no name at all. I think the Apple provider just returns an email.